plugins/ltac/g_rewrite.mlg

```diff
@@ -66,35 +66,26 @@ END
 
 {
 
-type raw_strategy = (constr_expr, Genredexpr.raw_red_expr) strategy_ast
-type glob_strategy = (glob_constr_and_expr, Genredexpr.glob_red_expr) strategy_ast
-
-let interp_strategy ist env sigma s =
-  let interp_redexpr r = fun env sigma -> Tacinterp.interp_red_expr ist env sigma r in
-  let interp_constr c = (fst c, fun env sigma -> Tacinterp.interp_open_constr ist env sigma c) in
-  let s = map_strategy interp_constr interp_redexpr s in
-  strategy_of_ast s
-
-let glob_strategy ist s = map_strategy (Tacintern.intern_constr ist) (Tacintern.intern_red_expr ist) s
+(** XXX this seems incorrect? *)
 let subst_strategy s str = str
```

Comment:
> * nothing elsewhere

Don't you mean "and recursively walk the term elsewhere"?

Reply:
@herbelin This errors with
```
Error: This expression has type
         Mod_subst.substitution ->
         (Constr.constr, Redexpr.red_expr, 'a) Rewrite.strategy_ast ->
         (Constr.constr, Redexpr.red_expr, 'a) Rewrite.strategy_ast
       but an expression was expected of type
         Ltac_plugin.Tacexpr.glob_strategy Gensubst.subst_fun =
           Mod_subst.substitution ->
           Ltac_plugin.Tacexpr.glob_strategy ->
           Ltac_plugin.Tacexpr.glob_strategy
       Type (Constr.constr, Redexpr.red_expr, 'a) Rewrite.strategy_ast
       is not compatible with type
         Ltac_plugin.Tacexpr.glob_strategy =
           (Genintern.glob_constr_and_expr, Genredexpr.glob_red_expr,
            Names.Id.t)
           Rewrite.strategy_ast
       Type Constr.constr = Constr.t is not compatible with type
         Genintern.glob_constr_and_expr =
           Glob_term.glob_constr * Constrexpr.constr_expr option
```

Reply:
See commented out code in the latest commit

Reply:
Btw, why are
https://github.com/coq/coq/blob/973cc473d47f5fe9e794c077770c3a085b8cfe3e/plugins/ltac/tacsubst.mli#L28-L33
in plugins/ltac/tacsubst rather than in genintern where `glob_constr_and_expr` is defined, or perhaps in a new gensubst?

Reply:
You're right, use `subst_glob_constr_and_expr`.

My understanding is that the general preference would rather be to move `glob_constr_and_expr` to `plugins/ltac`.

Reply:
Figured it out, but still curious about the misc section of tacsubst.

Reply:
Fixed